source/write/bulk-write.txt

```diff
@@ -0,0 +1,344 @@
+     - | Provides a map of parameter names and values to set top-level
+         variables for the operation. Values must be constant or closed
+         expressions that don't reference document fields.
+
+The following code creates options and uses the ``ordered(false)`` option to
+specify an unordered bulk write. Then, the example uses the
+``bulkWrite()`` method to perform a bulk operation:
+
+.. literalinclude:: /includes/write/bulk.kt
+   :start-after: start-bulk-write-unordered
+   :end-before: end-bulk-write-unordered
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+If any of the write operations in an unordered bulk write fail, the {+driver-short+}
+reports the errors only after attempting all operations.
+
+.. note::
+
+   Unordered bulk operations do not guarantee an order of execution. The
+   order can differ from the way you list them to optimize the runtime.
+
+Return Value
+------------
+
+The ``bulkWrite()`` method returns a ``BulkWriteResult`` object. You can
+access the following information from a ``BulkWriteResult`` instance:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``wasAcknowledged()``
+     - | Indicates if the server acknowledged the write operation.
+         
+   * - ``getDeletedCount()``
+     - | The number of documents deleted, if any.
+
+   * - ``getInsertedCount()``
+     - | The number of documents inserted, if any.
+
+   * - ``getInserts()``
+     - | The list of inserted documents, if any.
+
+   * - ``getMatchedCount()``
+     - | The number of documents matched for an update, if applicable.
+
+   * - ``getModifiedCount()``
+     - | The number of documents modified, if any.
+
+   * - ``getUpserts()``
+     - | The list of upserted documents, if any.
+
+Additional Information
+----------------------
+
+To learn how to perform individual write operations, see the following guides:
+
+- :ref:`kotlin-sync-write-insert`
+- :ref:`kotlin-sync-write-update`
+- :ref:`kotlin-sync-write-delete`
+
+.. - :ref:`kotlin-sync-write-replace`
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+To learn more about any of the methods or types discussed in this
+guide, see the following API Documentation:
+
+- `bulkWrite() <{+api+}/mongodb-driver-kotlin-sync/com.mongodb.kotlin.client/-mongo-collection/bulk-write.html>`__
+- `DeleteManyModel <{+core-api+}/com/mongodb/client/model/DeleteManyModel.html>`__
+- `DeleteOneModel <{+core-api+}/com/mongodb/client/model/DeleteOneModel.html>`__
+- `InsertOneModel <{+core-api+}/com/mongodb/client/model/InsertOneModel.html>`__
+- `ReplaceOneModel <{+core-api+}/com/mongodb/client/model/ReplaceOneModel.html>`__
+- `UpdateManyModel <{+core-api+}/com/mongodb/client/model/UpdateManyModel.html>`__
+- `UpdateOneModel <{+core-api+}/com/mongodb/client/model/UpdateOneModel.html>`__
+- `BulkWriteOptions <{+core-api+}/com/mongodb/client/model/BulkWriteOptions.html>`__
+- `BulkWriteResult <{+core-api+}/com/mongodb/bulk/BulkWriteResult.html>`__
```

Comment:
Complete nitpick here, but I think we should list the singular operations before the many operations.
```suggestion
- `bulkWrite() <{+api+}/mongodb-driver-kotlin-sync/com.mongodb.kotlin.client/-mongo-collection/bulk-write.html>`__
- `DeleteOneModel <{+core-api+}/com/mongodb/client/model/DeleteOneModel.html>`__
- `DeleteManyModel <{+core-api+}/com/mongodb/client/model/DeleteManyModel.html>`__
- `InsertOneModel <{+core-api+}/com/mongodb/client/model/InsertOneModel.html>`__
- `ReplaceOneModel <{+core-api+}/com/mongodb/client/model/ReplaceOneModel.html>`__
- `UpdateOneModel <{+core-api+}/com/mongodb/client/model/UpdateOneModel.html>`__
- `UpdateManyModel <{+core-api+}/com/mongodb/client/model/UpdateManyModel.html>`__
- `BulkWriteOptions <{+core-api+}/com/mongodb/client/model/BulkWriteOptions.html>`__
- `BulkWriteResult <{+core-api+}/com/mongodb/bulk/BulkWriteResult.html>`__
```

Reply:
SUre, this was alphabetical but I will change